src/pages/status/SetupSteps/ChequebookDeployFund.tsx

```diff
@@ -1,38 +1,22 @@
 import { Typography } from '@material-ui/core/'
-import { CheckCircle, Warning } from '@material-ui/icons/'
 import EthereumAddress from '../../../components/EthereumAddress'
 import DepositModal from '../../../components/DepositModal'
 import CodeBlockTabs from '../../../components/CodeBlockTabs'
-import type { ChequebookAddressResponse, ChequebookBalanceResponse } from '@ethersphere/bee-js'
 import type { ReactElement } from 'react'
 
-interface Props {
-  chequebookAddress: ChequebookAddressResponse | null
-  chequebookBalance: ChequebookBalanceResponse | null
-  isLoadingChequebookAddress: boolean
-  isLoadingChequebookBalance: boolean
-}
+type Props = StatusChequebookHook
 
-const ChequebookDeployFund = (props: Props): ReactElement => (
-  <div>
-    <p style={{ marginBottom: '20px', display: 'flex' }}>
-      <span style={{ marginRight: '40px' }}>Deploy chequebook and fund with BZZ</span>
-      {props.chequebookAddress?.chequebookaddress ? <DepositModal /> : null}
-    </p>
-    <div style={{ marginBottom: '10px' }}>
-      {
-        // FIXME: this should be broken up
-        /* eslint-disable no-nested-ternary */
-        props.chequebookAddress?.chequebookaddress &&
-        props.chequebookBalance &&
-        props.chequebookBalance?.totalBalance > 0 ? (
-          <div>
-            <CheckCircle style={{ color: '#32c48d', marginRight: '7px', height: '18px' }} />
-            <span>Your chequebook is deployed and funded!</span>
-          </div>
-        ) : props.isLoadingChequebookAddress || props.isLoadingChequebookBalance ? null : (
+const ChequebookDeployFund = ({ isLoading, chequebookAddress, chequebookBalance }: Props): ReactElement | null => {
+  if (isLoading) return null
+
+  return (
+    <div>
+      <p style={{ marginBottom: '20px', display: 'flex' }}>
+        {chequebookAddress?.chequebookaddress && <DepositModal />}
+      </p>
+      <div style={{ marginBottom: '10px' }}>
+        {!(chequebookAddress?.chequebookaddress && chequebookBalance && chequebookBalance?.totalBalance > 0) && (
```

Comment:
I find this logical condition a bit hard to follow. Maybe it would be worth extracting some values to variables to simplify.

Reply:
Yeah these should be simple components that just display values that are always defined. But such rewrite seemed beyond the scope what we are doing now :)